src/llama-context.cpp

```diff
@@ -309,6 +310,40 @@ llama_context::llama_context(
                 throw std::runtime_error("failed to allocate compute pp buffers");
             }
 
+            if (cparams.flash_attn_type == LLAMA_FLASH_ATTN_TYPE_AUTO) {
+                bool fa_backend_mismatch = false;
+                GGML_ASSERT(ggml_graph_node(gf, 0)->op != GGML_OP_FLASH_ATTN_EXT);
+                for (int i = 1; i < ggml_graph_n_nodes(gf); i++) {
```

Comment:
I don't think this is a good way to do it. It is very fragile code that makes a lot of assumptions that are not guaranteed anywhere, and will break very easily and in a very difficult way to detect when making changes to other parts of the code.